plugins/nf-validation/src/main/nextflow/validation/SchemaValidator.groovy

```diff
@@ -172,52 +152,28 @@ class SchemaValidator extends PluginExtensionPoint {
             throw new SchemaValidationException("", [])
         }
         else if (samplesheetValue.containsKey('schema')) {
-            schemaFile = Path.of(getSchemaPath(baseDir, samplesheetValue['schema'].toString()))
+            schemaFile = Path.of(Utils.getSchemaPath(baseDir, samplesheetValue['schema'].toString()))
         } else {
             log.error "Parameter '--$samplesheetParam' does not contain a schema in the parameter schema ($schemaFilename). Unable to create a channel from it."
             throw new SchemaValidationException("", [])
         }
 
-        log.debug "Starting validation: '$samplesheetFile' with '$schemaFile'"
-
-        // Validate samplesheet
-        def String fileType = SamplesheetConverter.getFileType(samplesheetFile)
-        def String delimiter = fileType == "csv" ? "," : fileType == "tsv" ? "\t" : null
-        def List<Map<String,String>> fileContent
-        def Boolean s3PathCheck = params.validationS3PathCheck ? params.validationS3PathCheck : false
-        def Map types = variableTypes(schemaFile.toString(), baseDir)
-        def Boolean containsHeader = !(types.keySet().size() == 1 && types.keySet()[0] == "")
+        // Convert to channel
+        final channel = CH.create()
+        def List arrayChannel = []
+        try {
+            arrayChannel = SamplesheetConverter.convertToList(samplesheetFile, schemaFile)
+        } catch (Exception e) {
+            log.error(
+                """ Following error has been found during samplesheet conversion:
+    ${e}
 
-        if(!containsHeader){
-            types = ["empty": types[""]]
-        }
-        if(fileType == "yaml"){
-            fileContent = new Yaml().load((samplesheetFile.text)).collect {
-                if(containsHeader) {
-                    return it as Map
-                }
-                return ["empty": it] as Map
-            }
-        }
-        else if(fileType == "json"){
-            fileContent = new JsonSlurper().parseText(samplesheetFile.text).collect {
-                if(containsHeader) {
-                    return it as Map
-                }
-                return ["empty": it] as Map
-            }
-        }
-        else {
-            fileContent = samplesheetFile.splitCsv(header:containsHeader ?: ["empty"], strip:true, sep:delimiter, quote:'\"')
-        }
-        def List<Map<String,String>> fileContentCasted = castToType(fileContent, types)
-        if (validateFile(false, samplesheetFile.toString(), fileContentCasted, schemaFile.toString(), baseDir, s3PathCheck)) {
-            log.debug "Validation passed: '$samplesheetFile' with '$schemaFile'"
+Please run validateParameters() first before trying to convert a samplesheet to a channel.
+Reference: https://nextflow-io.github.io/nf-validation/parameters/validation/
+""" as String
+            )
```

Comment:
Ooh is this a change in functionality? Does samplesheet converter no longer do a second validation?

Reply:
Nope we decided to not do that anymore and instead give this error. It's a bit stupid to do the validation twice (even though it works pretty fast with the new library)